### ISSUE TITLE
com.df.discoveryType: DNS doesn't work with global services

### ISSUE THREAD

**Issue by jaschaio** [docker-flow/docker-flow-proxy#21]
I am using the new `com.df.discoveryType: DNS` as described in #13 

Unfortunately it doesn't work when the service is set to `mode: global`. I have to use `replicated` mode and set a specific number of `replicas` to get it to work.

If I try it with `mode: global` I get the Following error:

```
2018-05-24T10:54:15.821057657Z proxy_proxy.0.kvxmdwqiqg57@swarm-manager-01    | 2018/05/24 10:54:15 Validating configuration
2018-05-24T10:54:15.828949479Z proxy_proxy.0.kvxmdwqiqg57@swarm-manager-01    | [ALERT] 143/105415 (265) : parsing [/cfg/haproxy.cfg:87] : Wrong server-template number or range arg '0'.
2018-05-24T10:54:15.836071732Z proxy_proxy.0.kvxmdwqiqg57@swarm-manager-01    | [ALERT] 143/105415 (265) : Error(s) found in configuration file : /cfg/haproxy.cfg
2018-05-24T10:54:15.836131332Z proxy_proxy.0.kvxmdwqiqg57@swarm-manager-01    | [ALERT] 143/105415 (265) : Fatal errors found in configuration.
2018-05-24T10:54:15.850056632Z proxy_proxy.0.kvxmdwqiqg57@swarm-manager-01    | Exit Status: 1
```

The generated Backend looks like this:
```
backend app_web-be80_0
     mode http
     http-request redirect scheme https if !{ ssl_fc }
     http-request add-header X-Forwarded-Proto https if { ssl_fc }
     server-template app_web 0 app_web:80 check
```

This is probably intended. Not sure if there is an easy way for docker flow proxy to get the number of replicas running for a global service. Just wanted to make sure that this is not a bug.

**Comment by thomasjpfan** [comment]
I think this is a bug. `global` services should be supported.

There are two quick ways to fix this bug:

1. Update DFSL to use the `replicas`parameter to mean "the number of tasks running".
2. Update DFP to use `net.LookupHost` to get the number of tasks running when `replicas==0`.

@vfarcic What do you think?

**Comment by thomasjpfan** [comment]
I went with option 2. 

@jaschaio This feature is available at `dockerflow/docker-flow-proxy:18.05.26-51`. Can you test this release out and see if it works for your use case?

**Comment by vfarcic** [comment]
@jaschaio I'll assume that the fix done by @thomasjpfan solved this issue and close it. Please reopen if you're still experiencing the same problem.
